packages/ui-components/Common/Select/SelectScrollButton/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/24/outline';
+import { useEffect, useRef, useState } from 'react';
+import { type FC, type RefObject } from 'react';
+
+import styles from '@node-core/ui-components/Common/Select/index.module.css';
+
+type SelectScrollButtonProps = {
+  direction: 'up' | 'down';
+  selectContentRef?: RefObject<HTMLDivElement | null>;
+  scrollAmount?: number;
+  scrollInterval?: number;
+};
+
+const SelectScrollButton: FC<SelectScrollButtonProps> = ({
+  direction,
+  selectContentRef,
+  scrollAmount = 35,
+  scrollInterval = 50,
+}) => {
+  const DirectionComponent =
+    direction === 'down' ? ChevronDownIcon : ChevronUpIcon;
+  const [isVisible, setIsVisible] = useState(false);
+  const [hasOverflow, setOverflow] = useState(false);
+  const intervalRef = useRef<number | null>(null);
+  const isScrollingRef = useRef(false);
+
+  const clearScrollInterval = () => {
+    if (intervalRef.current !== null) {
+      window.clearInterval(intervalRef.current);
+      intervalRef.current = null;
+    }
+  };
+
+  const startScrolling = () => {
+    if (!selectContentRef?.current || !isVisible || !hasOverflow) return;
+
+    clearScrollInterval();
+
+    intervalRef.current = window.setInterval(() => {
+      if (!selectContentRef.current || !isScrollingRef.current) return;
+
+      const container = selectContentRef.current;
+
+      if (direction === 'down') {
+        container.scrollBy({ top: scrollAmount, behavior: 'smooth' });
+
+        if (
+          container.scrollTop >=
+          container.scrollHeight - container.clientHeight
+        ) {
+          clearScrollInterval();
+          setIsVisible(false);
+        }
+      } else {
+        container.scrollBy({
+          top: -Math.abs(scrollAmount),
+          behavior: 'smooth',
+        });
+
+        if (container.scrollTop <= 0) {
+          clearScrollInterval();
+          setIsVisible(false);
+        }
+      }
```

Comment:
This is an extremely extremely minor nitpick, but this could probably be simplified into a helper function?

(But this is such a tiny nitpick that you really don't have to do anything)